src/api/java/cvc5/Stat.java

```diff
@@ -0,0 +1,169 @@
+ * The value type can be queried (using `isInt()`, `isDouble()`, etc.) and
+ * the stored value can be accessed (using `getInt()`, `getDouble()`, etc.).
+ * It is possible to query whether this statistic is an expert statistic by
+ * `isExpert()` and whether its value is the default value by `isDefault()`.
+ */
+public class Stat extends AbstractPointer
+{
+   // region construction and destruction
+   Stat(Solver solver, long pointer)
+   {
+     super(solver, pointer);
+   }
+
+   protected static native void deletePointer(long pointer);
+
+   public long getPointer()
+   {
+     return pointer;
+   }
+
+   @Override
+   public void finalize()
+   {
+     deletePointer(pointer);
+   }
+
+   // endregion
+
+  /**
+   * @return a string representation of this Stat
+   */
+  protected String toString(long pointer)
+  {
+    throw new UnsupportedOperationException("Not implemented in the cpp api");
+  }
+
+  /**
+   * Is this value intended for experts only?
+   * @return Whether this is an expert statistic.
+   */
+  public boolean isExpert()
+  {
+    return isExpert(pointer);
+  }
+
+  private native boolean isExpert(long pointer);
+
+  /**
+   * Does this value hold the default value?
+   * @return Whether this is a defaulted statistic.
+   */
+  public boolean isDefault()
+  {
+    return isDefault(pointer);
+  }
+
+  private native boolean isDefault(long pointer);
+
+  /**
+   * Is this value an integer?
+   * @return Whether the value is an integer.
+   */
+  public boolean isInt()
+  {
+    return isInt(pointer);
+  }
+
+  private native boolean isInt(long pointer);
+
+  /**
+   * Return the integer value.
+   * @return The integer value.
+   */
+  public long getInt()
+  {
+    return getInt(pointer);
+  }
+
+  private native long getInt(long pointer);
+
+  /**
+   * Is this value a double?
+   * @return Whether the value is a double.
+   */
+  public boolean isDouble()
+  {
+    return isDouble(pointer);
+  }
+
+  private native boolean isDouble(long pointer);
+
+  /**
+   * Return the double value.
+   * @return The double value.
+   */
+  public double getDouble()
+  {
+    return getDouble(pointer);
+  }
+
+  private native double getDouble(long pointer);
+
+  /**
+   * Is this value a string?
+   * @return Whether the value is a string.
+   */
+  public boolean isString()
+  {
+    return isString(pointer);
+  }
+
+  private native boolean isString(long pointer);
+
+  /**
+   * Return the string value.
+   * @return The string value.
+   */
+  public String getString()
+  {
+    return getString(pointer);
+  }
+
+  private native String getString(long pointer);
+
+  /**
+   * Is this value a histogram?
+   * @return Whether the value is a histogram.
+   */
+  public boolean isHistogram()
+  {
+    return isHistogram(pointer);
+  }
+
+  private native boolean isHistogram(long pointer);
+
+  /**
+   * Return the histogram value.
+   * @return The histogram value.
+   */
+  public Map<String, Long> getHistogram()
```

Comment:
It is probably not a big deal here, but technically the types don't match as the C++ api uses `uint64_t` whereas `Long` corresponds to `int64_t`. How did you handle this in other places?
Maybe, we should explicitly safeguard this cast in the JNI code?

Reply:
I thought of using BigInteger here, but since this is a histogram, there is no precision loss here. The user can use functions like 
https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#toUnsignedString-long-